reactioncommerce/reaction
# Performance/Refactor remove revision control

Resolves #4237   
Impact: **breaking**  
Type: **performance/refactor**

## Issue
Remove the `Revision` control system in favor of the` Catalog` as having both is redundant.

## Solution

Remove all code, hooks, collections, and packages related to revision control.

## Breaking changes

- The revision control system is removed, any custom packages that depend on it will no longer work.
- removed `applyProductRevision` from `lib/api/products.js`
- Cannot undo changes when editing a product
- PDP page will have an instant reflection of changes. Product grid will only see changes upon publishing.

## Testing
1. As an admin
2. Add products, see that the changes show up immediately on PDP for customer and admin
3. Check the product grid has no changes for the customer, but immediate change for admin
4. Publish product, see product grid change for customers
5. Add products to cart and checkout
6. Fulfill orders and verify that inventory updates on the product via the PDP admin tools
